client/app/scripts/superdesk-archive/directives.js

```diff
@@ -323,15 +328,63 @@ define([
                                 } else {
                                     figure.html(img);
                                 }
+                                _calcRatio();
                             };
 
                             img.onerror = function() {};
                             img.src = href;
                         }
                     });
+
+                    var stopRatioWatch = scope.$watch('ratio', function(val) {
```

Comment:
seems complicated, if you have ratio set on scope from sdRatioCalc, why not use `ng-class="{portrait: ratio > 1.0}"` ?